### PR TITLE
Creation and Automation of the Donations Progress Bar

### PR DESCRIPTION
**Pull request by etiennecollin**
**This PR superseeds the [Donation Bar PR (#35)](https://github.com/aihelpsukraine/aihelpsukraine.github.io/pull/35).**
It is up-to-date with PR #35 and adds the automated part.

New features added:
- Donation bar is now working.
- The current donated amount shown by the bar is automatically updated from a google sheet.
- The google sheet's URL and the donation target may now be set via variables in the `/data/settings.yml` file.

### CODE REVIEW

**Comment by etiennecollin** [comment]
Made modifications. Now ready to reopen PR.

File: _data/settings.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
 talk:
-    isLive: False
-    speaker: Hannah Kerner
+  isLive: False
+  speaker: Hannah Kerner
+
+donations:
+  spreadsheetURL: https://docs.google.com/spreadsheets/d/e/2PACX-1vTXKpZ8iB5Nz6mHl5iYUAA68DU1D-gCnwuDREnVULI9JrOnh1CF8U1RF6zDbVfrkw/pub?gid=1443797343&single=true&output=csv
+  target: 100000
+  targetTextForm: "100,000.00"
```
Comment (alexhernandezgarcia) [inline _data/settings.yml:8]: I would remove the floating part, to simplify the reading.

File: _layouts/main.html [hunk 2]
```diff
@@ -1,26 +1,33 @@
...
+	</head>
+	<body>
+		<section class="hero is-fullheight">
+			{% include navigation.html icon-size="lg" %}
+			<div class="hero-body">
+				<div class="container">{{ content }}</div>
+			</div>
+			{% include footer.html icon-size="lg" %}
+		</section>
+	</body>
```
Comment (alexhernandezgarcia) [inline _layouts/main.html:32]: I can't easily see the actual changes because of the new indent. Would it be possible to revert the indent to 2 spaces?

**Review by alexhernandezgarcia — COMMENTED** [review]
Nice work! It would have taken me ages to write the parsing code, so thanks! :)

A couple of things:
* We need to wait before merging to confirm the donations from organisations.
* My comments are minor, but please take a look.

File: _data/settings.yml (hunk 1 above)
Comment (etiennecollin) [reply _data/settings.yml:8]: You mean you would write it as 100 000 such that it appears as "100 000 CAD" in the text?

File: _layouts/main.html (hunk 2 above)
Comment (etiennecollin) [reply _layouts/main.html:32]: No changes were made. The file was automatically formatted when I opened it. I will format it back to 2 spaces.

**Review by liellnima — COMMENTED** [review]
Looks all good to me, tested it - thank you so much for automating that :D

Only thing we need to wait for: 1M or 100k as target?

And we only wanted to have Ivado live as donor when it's finally confirmed. Can we add a "live" tag here as well, so we can already add our tentative donors in the code and put them live as soon as the money was received?

**Comment by etiennecollin** [comment]
> And we only wanted to have Ivado live as donor when it's finally confirmed. Can we add a "live" tag here as well, so we can already add our tentative donors in the code and put them live as soon as the money was received?

@liellnima If I understand what you want... would a "live" or "confirmed" switch for every donation in the google sheet be okay? Then the donations would only be computed in the total if the checkbox is checked? Or do you mean a switch to show/hide the "donor" cards?

**Comment by liellnima** [comment]
I initially thought only about the donor cards, but I really like your idea of handling that with the "confirmed" box in the spreadsheet.

TBH we can already show it in the donation bar even if it's not officially confirmed. We just don't want to show their logos because it's the benefit they are getting (visibility) and we only want to give them when we actually received the money

**Comment by liellnima** [comment]
I.e. if you can use the "is-confirmed" from spreadsheet as switch for the donor cards that's perfect, no need to adapt the donation bar based on confirmed/not-confirmed

**Comment by etiennecollin** [comment]
Right now, the "isConfirmed" is in `/_data/donors.yml` to alter the visibility of the cards. It is really hard (or I didn't find how) to use google sheets elements as YML variables and/or Jekyll Fluid conditions... I will keep it in the `/_data/donors.yml` and we can use the checkbox in the sheet if we really want to hide certain donations from the progress bar.

**Review by liellnima — APPROVED** [review]
good to go!

**Pull request merged**